app/gui2/src/stores/graph/graphDatabase.ts

```diff
@@ -224,7 +225,7 @@ export class GraphDb {
   })
 
   getNodeFirstOutputPort(id: NodeId): AstId {
-    return set.first(this.nodeOutputPorts.lookup(id)) ?? id
+    return set.first(this.nodeOutputPorts.lookup(id)) ?? this.idFromExternal(id)
```

Comment:
This can now return nullish if the input `NodeId` is for a node that hasn't been committed. The typechecker doesn't catch it because `set.first` is wrongly-typed as infallible.